Full-qualified SSM parameters not working

When using a SSM parameter name for latest ami-id with a leading `/` the policy is invalid.


Passing `gh-runner/ami/id` as a `amiIdSsmPath` with the `ImagePipeline` props will (correctly) fail with:
> [ERROR] ClientError: An error occurred (ValidationException) when calling the PutParameter operation: Parameter name must be a fully qualified name.

Therefore a nested SSM parameter must be passed with a leading slash (`/…`). Doing so will generate a invalid policy attached to the lambda role though:

```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Action": [
                "ssm:AddTagsToResource",
                "ssm:GetParameter",
                "ssm:GetParameterHistory",
                "ssm:GetParameters",
                "ssm:PutParameter"
            ],
            "Resource": "arn:aws:ssm:eu-central-1:123456789012:parameter//gh-runner/ami/id",
            "Effect": "Allow"
        }
    ]
}
```

whereas the correct policy would need to be slightly changed
```diff
--- AmiSsmUpdateLambdaPolicy
+++ AmiSsmUpdateLambdaPolicy
@@ -12,5 +12,5 @@
                 "ssm:PutParameter"
            ],
-           "Resource": "arn:aws:ssm:eu-central-1:123456789012:parameter//gh-runner/ami/id",
+           "Resource": "arn:aws:ssm:eu-central-1:123456789012:parameter/gh-runner/ami/id", 
            "Effect": "Allow"
         }
```

--- a/src/index.ts
+++ b/src/index.ts
@@ -291,10 +291,11 @@ export class ImagePipeline extends Construct {
      * (only if a Parameter Store path is provided)
      */
     if (props.amiIdSsmPath) {
+      const amiIdSsmPath = props.amiIdSsmPath.replace(/^\/+/, '/');
       const amiSsmUpdateLambdaPolicy = new iam.PolicyDocument({
         statements: [
           new iam.PolicyStatement({
-            resources: [`arn:aws:ssm:${props.amiIdSsmRegion}:${props.amiIdSsmAccountId}:parameter/${props.amiIdSsmPath}`],
+            resources: [`arn:aws:ssm:${props.amiIdSsmRegion}:${props.amiIdSsmAccountId}:parameter${amiIdSsmPath}`],
             actions: [
               'ssm:PutParameter',
               'ssm:GetParameterHistory',
@@ -320,7 +321,7 @@ export class ImagePipeline extends Construct {
         handler: 'image-builder-lambda-update-ssm.lambda_handler',
         role: amiSsmUpdateLambdaRole,
         environment: {
-          SSM_PATH: props.amiIdSsmPath,
+          SSM_PATH: amiIdSsmPath,
         },
         memorySize: 256,
       });

--- a/test/imagepipeline.test.ts
+++ b/test/imagepipeline.test.ts
@@ -109,8 +109,29 @@ test('Infrastructure Configuration IAM Role and Instance Profile are created', (
 });
 
 test('IAM Role contains necessary permission set', () => {
-  template.hasResourceProperties('AWS::IAM::Role',
-    Match.anyValue());
+  template.hasResourceProperties('AWS::IAM::Role', {
+    Policies: [
+      {
+        PolicyName: 'AmiSsmUpdateLambdaPolicy',
+        PolicyDocument: {
+          Statement: [
+            {
+              Effect: 'Allow',
+              Action: [
+                'ssm:PutParameter',
+                'ssm:GetParameterHistory',
+                'ssm:GetParameter',
+                'ssm:GetParameters',
+                'ssm:AddTagsToResource',
+              ],
+              Resource: 'arn:aws:ssm:us-east-1:11223344556:parameter/ec2-image-builder/al2-x86',
+            },
+          ],
+        },
+      },
+    ],
+  });
+
 });
 
 test('Infrastructure Configuration has the default instance types', () => {

